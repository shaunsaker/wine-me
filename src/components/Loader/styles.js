import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {
    alignSelf: 'stretch',
    height: styleConstants.dimensions.placeCard.height,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default styles;
