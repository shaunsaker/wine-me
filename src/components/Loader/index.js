import React from 'react';
import { View, ActivityIndicator } from 'react-native';

import styleConstants from '../../styleConstants';

import styles from './styles';

const propTypes = {};

const defaultProps = {};

const Loader = () => {
  return (
    <View style={styles.container}>
      <ActivityIndicator size="large" color={styleConstants.colors.primary} />
    </View>
  );
};

Loader.propTypes = propTypes;
Loader.defaultProps = defaultProps;

export default Loader;
