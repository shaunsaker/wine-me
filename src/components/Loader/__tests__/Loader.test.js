import React from 'react';
import renderer from 'react-test-renderer';

import Loader from '..';

describe('Loader', () => {
  const handlePress = jest.fn();

  describe('renders', () => {
    it('renders', () => {
      const component = renderer.create(<Loader handlePress={handlePress} />);

      expect(component).toMatchSnapshot();
    });
  });
});
