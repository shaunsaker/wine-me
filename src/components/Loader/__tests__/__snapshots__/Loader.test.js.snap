// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "height": 200,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#3D84A8"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
